CoreScriptsRoot/CoreScripts/Topbar.lua

```diff
@@ -1403,10 +1433,183 @@ local function CreateStopRecordIcon()
+
+	function menuItem:OnClicked(wasActive)
+		if wasActive then
+			VRHub:FireModuleClosed(thisModuleName)
+			return
+		else
+			VRHub:FireModuleOpened(thisModuleName, true, true, true)
+		end
+
+		spawn(function()
+			menuItem:SetActive(true)
+		
+			countdownPanel:SetVisible(true)
+
+			for i = 3, 1, -1 do
+				if menuItem:IsActive() then
+					countdown.Text = tostring(i)
+					wait(1)
+				end
+			end
+
+			countdownPanel:SetVisible(false)			
+
+			if menuItem:IsActive() then
+				InputService:RecenterUserHeadCFrame() 
+			end
+			VRHub:FireModuleClosed(thisModuleName)
+		end)
+	end
+
+	return menuItem
+end
+
+-------------------
+
+---- Notifications VR -- 
+
+local function CreateNotificationsIcon3D(topBarInstance, panel, menubar)
+--Awaiting commit of NotificationHub.lua
+	local VRHub = require(GuiRoot.Modules.VR.VRHub)
+	local thisModuleName = "Notifications"
+
+	local menuItem = CreateMenuItem3D(thisModuleName)
+	menuItem:SetHoverText("Notifications")
+
+--[[	local NotificationHubModule = require(GuiRoot.Modules.VR.NotificationHub)
+
+	
+
+	local icon = Util.Create "ImageLabel" {
+		Parent = menuItem:GetInstance(),
+
+		Position = UDim2.new(0.5, -12, 0.5, -16),
+		Size = UDim2.new(0, 24, 0, 32),
+
+		BackgroundTransparency = 1,
+
+		Image = "rbxasset://textures/ui/VR/notifications.png"
+	}
+
+	local notificationBubble = Util.Create "ImageLabel" {
+		Parent = menuItem:GetInstance(),
+
+		BackgroundTransparency = 1,
+		
+		Position = UDim2.new(0.5, 0, 0.25, 0),
+		Size = UDim2.new(0, 18, 0, 18),
+
+		Image = "rbxasset://textures/ui/Chat/MessageCounter.png",
+		ScaleType = Enum.ScaleType.Slice,
+		SliceCenter = Rect.new(9, 0, 9, 18),
+
+		Visible = false
+	}
+
+	local notificationCount = Util.Create "TextLabel" {
+		Parent = notificationBubble,
+
+		BackgroundTransparency = 1,
+
+		Text = "0",
+		TextColor3 = Color3.new(1, 1, 1),
+		Font = Enum.Font.SourceSansBold,
+		FontSize = Enum.FontSize.Size14,
+
+		Position = UDim2.new(0, -1, 0, -2),
+		Size = UDim2.new(1, 0, 1, 0)
+	}
+
+	local function setCounter(number)
+		if number == 0 then
+			notificationBubble.Visible = false
+		else
+			notificationBubble.Visible = true
+		end
+		notificationCount.Text = tostring(number)
+		notificationBubble.Size = UDim2.new(0, notificationCount.TextBounds.X + 11, 0, 18)
+	end
+	NotificationHubModule.UnreadCountChanged = setCounter
+
+	function menuItem:OnClicked(wasActive)
+		menuItem:SetActive(not wasActive)
+		NotificationHubModule:SetVisible(not wasActive)
+	end
+
+	VRHub.ModuleClosed.Event:connect(function(moduleName)
+		if moduleName == thisModuleName then
+			menuItem:SetActive(false)
+		end
+	end)
+	]]
+
+	return menuItem
+end
+------------------------
+
 local TopBar = CreateTopBar()
 local LeftMenubar = CreateMenuBar(BarAlignmentEnum.Left)
 local RightMenubar = CreateMenuBar(BarAlignmentEnum.Right)
-local ThreeDMenubar = Create3DMenuBar(BarAlignmentEnum.Left, Topbar3DPanel)
+local Menubar3D = CreateMenuBar3D(BarAlignmentEnum.Left, TopbarPanel3D)
```

Comment:
I know this isn't your change, but it seems like we are mixing names here; we should decide to call it Topbar or Menubar